sdk/src/signer/mod.rs

```diff
@@ -94,29 +94,6 @@ where
     }
 }
 
-impl<Container: Deref<Target = impl Signer>> Signer for Container {
-    #[inline]
-    fn pubkey(&self) -> Pubkey {
-        self.deref().pubkey()
-    }
-
-    fn try_pubkey(&self) -> Result<Pubkey, SignerError> {
-        self.deref().try_pubkey()
-    }
-
-    fn sign_message(&self, message: &[u8]) -> Signature {
-        self.deref().sign_message(message)
-    }
-
-    fn try_sign_message(&self, message: &[u8]) -> Result<Signature, SignerError> {
-        self.deref().try_sign_message(message)
-    }
-
-    fn is_interactive(&self) -> bool {
-        self.deref().is_interactive()
-    }
-}
```

Comment:
Compiler doesn't seem to be complaining about me removing this, is this even used ? If it's dead code I'd rather remove it, appreciate feedback!

Reply:
Haha, I found out when I ran `./cargo test`, it's used to support wrappers like Box/Rc/Arc around `Signer`,

anyway, put it back and adjusted accordingly, also fixed failing tests in a [follow-up commit](https://github.com/solana-labs/solana/pull/33860/commits/b92cc16b632c09cef4faf2c3559339c38531d8a9).